lib/erb2epp/parser.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'temple'
+
+module Erb2epp
+  # Parse an ERB template to Temple S-expressions respecting the whitespace control
+  class Parser < Temple::Parser
+    ERB_PATTERN = /(\n|<%%|%%>)|<%(=|\#)?(.*?)?%>/m.freeze
+
+    def call(input)
+      result = [:multi]
+      pos = 0
+
+      input.scan(ERB_PATTERN) do |token, indicator, code|
+        lm = Regexp.last_match
```

Comment:
I wonder if [StringScanner](https://rubyapi.org/3.2/o/stringscanner) is a better API to achieve this.

Reply:
I tried with StringScanner w/o success though.. I was unable to make it work with the regexp.. maybe I miss something but 🤷🏻‍♂️